### ISSUE TITLE
Custom text ending with a Hangul word does not work

### ISSUE THREAD

**Issue by j824h** [monkeytypegame/monkeytype#4968]
### Did you clear cache before opening an issue?

- [X] I have cleared my cache

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Does the issue happen when logged in?

Yes

### Does the issue happen when logged out?

Yes

### Does the issue happen in incognito mode when logged in?

Yes

### Does the issue happen in incognito mode when logged out?

Yes

### Account name

j824h

### Account config

{"theme":"serika_dark","themeLight":"serika","themeDark":"serika_dark","autoSwitchTheme":false,"customTheme":false,"customThemeColors":["#323437","#e2b714","#e2b714","#646669","#2c2e31","#d1d0c5","#ca4754","#7e2a33","#ca4754","#7e2a33"],"favThemes":[],"showKeyTips":true,"showLiveWpm":false,"showTimerProgress":true,"smoothCaret":"medium","quickRestart":"off","punctuation":false,"numbers":false,"words":50,"time":15,"mode":"custom","quoteLength":[0,1,2,3],"language":"english_1k","fontSize":1.5,"freedomMode":false,"difficulty":"normal","blindMode":false,"quickEnd":false,"caretStyle":"default","paceCaretStyle":"default","flipTestColors":false,"layout":"default","funbox":"none","confidenceMode":"off","indicateTypos":"off","timerStyle":"mini","colorfulMode":false,"randomTheme":"off","timerColor":"main","timerOpacity":"1","stopOnError":"off","showAllLines":false,"keymapMode":"off","keymapStyle":"staggered","keymapLegendStyle":"lowercase","keymapLayout":"overrideSync","keymapShowTopRow":"layout","fontFamily":"Roboto_Mono","smoothLineScroll":false,"alwaysShowDecimalPlaces":false,"alwaysShowWordsHistory":true,"singleListCommandLine":"manual","capsLockWarning":true,"playSoundOnError":"off","playSoundOnClick":"off","soundVolume":"0.5","startGraphsAtZero":true,"showOutOfFocusWarning":true,"paceCaret":"off","paceCaretCustomSpeed":100,"repeatedPace":true,"pageWidth":"125","accountChart":["on","on","on","on"],"minWpm":"off","minWpmCustomSpeed":100,"highlightMode":"letter","typingSpeedUnit":"wpm","ads":"result","hideExtraLetters":false,"strictSpace":false,"minAcc":"off","minAccCustom":90,"showLiveAcc":false,"showLiveBurst":false,"monkey":false,"repeatQuotes":"off","oppositeShiftMode":"off","customBackground":"","customBackgroundSize":"cover","customBackgroundFilter":[0,1,1,1,1],"customLayoutfluid":"qwerty#dvorak#colemak","monkeyPowerLevel":"off","minBurst":"off","minBurstCustomSpeed":100,"burstHeatmap":false,"britishEnglish":false,"lazyMode":false,"showAverage":"off","tapeMode":"off"}

### Current Behavior

When custom text includes a line ending with a Hangul word, and I either correctly or incorrectly complete the word and type return, the word is cleared.

### Expected Behavior

When I type return, the cursor should go to the next line, with getting the new line character correct.

### Steps To Reproduce

1. Go to "Change custom text"
2. Set the custom text so that it includes a line ending with a Hangul word. Example:
```
abc 가나다
abc
```
3. Type either `abc 가나다<Return>` or `abc abc<Return>` or `abc ...<Return>`

### Environment

- OS: Windows 11
- Browser: Google Chrome
- Browser Version: 120.0.6099.225 (Official Build) (64-bit)

### Anything else?

For now, the problem can be circumvented by "Replace new lines with spaces".

**Comment by josehp304** [comment]
I tried it and i couldn't reproduce this.

Its's working normaly for me
![Uploading Capture.PNG…]()

**Comment by j824h** [comment]
@josehp304 Is your image working? I only see "Uploading Capture.PNG…"

Thanks for trying this out by yourself.
I would appreciate if I can see what has worked for you, as I still seem to have the issue.

**Comment by josehp304** [comment]
![Capture](https://github.com/monkeytypegame/monkeytype/assets/83821474/e4c841a2-389d-40ca-b51b-485a70fe1817)

this is what it showed me

**Comment by j824h** [comment]
@josehp304 Yep, that is showing exactly what I intended to report as a bug. We should be able to finish the line by a return, as skipping with a space and getting the word as incorrect.

**Comment by j824h** [comment]
This seems to be still in a same state. In addition, I'm adding some environments that I got to currently reproduce this bug.

### Environment

OS: macOS 15.2
Browser: Safari
Browser Version: Version 18.2 (20620.1.16.11.8)

OS: macOS 15.2
Browser: Google Chrome
Browser Version: Version 131.0.6778.205 (Official Build) (arm64)

**Comment by nadalaba** [comment]
@j824h monkeytype had a complete rewrite of the input system. Can you check if this is still an issue?

**Comment by j824h** [comment]
@nadalaba I checked again in monkeytype.com, and the behavior has changed.

I tried the same natural input pattern as before. When a custom text line ends with a Hangul word, if I press return while the final Hangul character is still mid-composition (IME), that character gets cleared. And this happening somehow affects the behavior even after the line ending with Hangul word.

![Image](https://github.com/user-attachments/assets/e674376e-2590-4296-adcb-1d21086cfb04)

Previously, the entire word would be wiped. Now only the in-progress final character disappears, and the preceding characters remain correct.

So the issue seems to be not fully resolved.

**Comment by j824h** [comment]
Also, there now seems to be another user-level workaround: if I press Right Arrow to commit the in-progress Hangul character, and then press return, the character isn’t cleared.

![Image](https://github.com/user-attachments/assets/630d7dbf-1659-4872-97d9-3069b60b9b70)

**Comment by Miodec** [comment]
@j824h could you check if everything is good here? https://monkey-type--composition-dslz9ngi.web.app

**Comment by j824h** [comment]
@Miodec Oh, yes! Everything is indeed good there. Checked in Safari and Google Chrome, macOS. If this was already merged or is going to be merged, please close this! 🚀

**Comment by Miodec** [comment]
> [@Miodec](https://github.com/Miodec) Oh, yes! Everything is indeed good there. Checked in Safari and Google Chrome, macOS. If this was already merged or is going to be merged, please close this! 🚀

Very nice, sorry it took a while. The new input system made this issue much easier to figure out.
